client/packages/system/src/Vaccination/Components/SelectItemAndBatch.tsx

```diff
@@ -32,6 +32,18 @@ export const SelectItemAndBatch = ({
     );
   }, [dose.id]);
 
+  const isHistorical = draft.date?.toDateString() !== new Date().toDateString();
+
+  const selectBatch =
+    !isHistorical || draft.createTransactions || hasExistingSelectedBatch;
+
+  // Auto-select if there is only one item (and not already selected)
+  useEffect(() => {
+    if (vaccineItemOptions.length === 1 && !draft.itemId) {
+      updateDraft({ itemId: vaccineItemOptions[0]!.value });
+    }
+  }, [vaccineItemOptions]);
```

Comment:
Added the autoselect to item as well, seemed weird to have one without the other? Can back out if you're judging me rn 😁